lib/internal/errors.js

```diff
@@ -851,6 +851,22 @@ class AbortError extends Error {
   }
 }
 
+// TAP Validation Error used by the TAP checker/parser.
+class TAPValidationError extends Error {
```

Comment:
If we coerce invalid TAP to a diagnostic line, why are these errors needed?

Reply:
These errors are used when the parser is used in non-streaming scenario:
```
const parser = new TapParser();
const ast = parser.parse(`TAP version 13`);
```

Do we still need to support that? Or do we want developers (if we export the TapParser) to only use the parser as a stream transformer? cc @cjihrig

Reply:
I don't feel strongly either way, but I could definitely picture the parser being used outside of the context of a stream.

Reply:
I copied the class construction from other parts of this file, assuming this was the right way for the parser. Mind sharing an example of error codes?

Reply:
Check out the bottom of this same file. All of the `E()` calls are creating error codes.

Reply:
Oh, I see. Actually, if you check [these lines](https://github.com/nodejs/node/blob/27ec84b81cc46ec7725e7bcc99e40c7f726a9457/lib/internal/errors.js#L1612-L1664), I am already using `E()` to show custom errors (line, column, and position) during scanning and parsing. I use these classes as a type. Do you recommend completely removing these custom class types and just returning an error message?

Reply:
Removed the 2 classes, and kept relevant error codes:
- `E('ERR_TAP_PARSER_ERROR', ...)` // extends SyntaxError
- `E('ERR_TAP_VALIDATION_ERROR', ...)` extends  Error
- `E('ERR_TAP_LEXER_ERROR', ...)` extends Error